### PR TITLE
more info about how to scrape k0s internals

### PR DESCRIPTION
**Pull request by fenio**
## Description

There was ongoing discussion in https://github.com/k0sproject/k0s/issues/2321 about how to scrape internals of k0s. 
No one in the end created actual PR so I decided to do it on my own as I hit exact same problem.

Fixes # (issue)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [X] Documentation update

## How Has This Been Tested?

- [X] Manual test
- [ ] Auto test added

I'm fine with rewriting it and do some changes as English is not my native language but please DO include some extra info about how to enable these metrics and scraping of them.

### CODE REVIEW

File: docs/system-monitoring.md [hunk 1]
```diff
@@ -11,6 +11,50 @@ You can read more about metrics for Kubernetes system components [here](https://
 sudo k0s install controller --enable-metrics-scraper
 ```
 
+Once you enable it new set of objects will show up on your cluster:
```
Comment (twz123) [inline docs/system-monitoring.md:14]:
```suggestion
Once enabled, a new set of objects will appear in the cluster:
```

File: docs/system-monitoring.md [hunk 2]
```diff
@@ -11,6 +11,50 @@ You can read more about metrics for Kubernetes system components [here](https://
...
+NAME                              READY   UP-TO-DATE   AVAILABLE   AGE
+deployment.apps/k0s-pushgateway   1/1     1            1           43h
+
+NAME                                         DESIRED   CURRENT   READY   AGE
+replicaset.apps/k0s-pushgateway-6c5d8c54cf   1         1         1       43h
+```
+
+That's not enough to start scraping these additional metrics.
+
+For Prometheus based solution you can create ServiceMonitor for it like this:
```
Comment (twz123) [inline docs/system-monitoring.md:33]:
```suggestion
That's not enough to start scraping these additional metrics. For Prometheus
Operator](https://prometheus-operator.dev/) based solutions, you can create a
`ServiceMonitor` for it like this:
```

File: docs/system-monitoring.md [hunk 3]
```diff
@@ -11,6 +11,50 @@ You can read more about metrics for Kubernetes system components [here](https://
...
+  - port: http
+  selector:
+    matchLabels:
+      app: k0s-observability
+      component: pushgateway
+      k0s.k0sproject.io/stack: metrics
+```
+
+Note that it won't clear alerts like "KubeControllerManagerDown" nor "KubeSchedulerDown" as they are based on Prometheus's internal "up" metrics.
+But you can get rid of these alerts by modifying them to detect working component like this:
```
Comment (twz123) [inline docs/system-monitoring.md:52]:
```suggestion
Note that it won't clear alerts like "KubeControllerManagerDown" or
"KubeSchedulerDown" as they are based on Prometheus' internal "up" metrics. But
you can get rid of these alerts by modifying them to detect a working component
like this:
```

**Review by twz123 — COMMENTED** [review]
Thanks for taking care of this!

Left some nits. Please also note the linter comments, and your commit needs to be signed-off.

**Comment by fenio** [comment]
> Thanks for taking care of this!
> 
> Left some nits. Please also note the linter comments, and your commit needs to be signed-off.

I've just accepted your suggestions. Is it now fine or do I have to do some extra to sign it off?
When I was accepting these changes there was "sign off" on a button so I guess it's enough?

**Comment by twz123** [comment]
There's still the first commit 809a7728abe7ba9f486fc271190fb5f8028e867e that's not signed off.

Also, the linter warnings on line 56 and 58 need to be addressed.

**Comment by fenio** [comment]
But after all I just wanted to distinguish it from rest of the text. If there's a better way to do it and not mark it as a code at the same time I'm happy with whatever solution.

**Comment by jnummelin** [comment]
I believe we've used _text_ as the marker in few other places.

To sign-off, I believe it would be easiest if you squash all the commits as one and then use something like `git commit -s --amend` to ad the signoff statement into the commit

**Comment by fenio** [comment]
I tried to sign off changes again but apparently I rebased and signed bunch of unrelated stuff.
Maybe it will be easier to start from scratch. I'll close this PR and create another one and sign it off properly.
Sorry for the mess.

**Pull request closed without merging**